scrapy/settings/__init__.py

```diff
@@ -110,6 +139,25 @@ def setmodule(self, module, priority='project'):
             if key.isupper():
                 self.set(key, getattr(module, key), priority)
 
+    def update(self, values, priority='project'):
+        self._assert_mutability()
+        if values is not None:
+            if isinstance(values, BaseSettings):
+                for name, value in six.iteritems(values):
+                    self.set(name, value, values.getpriority(name))
+            else:
+                for name, value in six.iteritems(values):
+                    self.set(name, value, priority)
+
+    def delete(self, name, priority='project'):
+        self._assert_mutability()
+        priority = get_settings_priority(priority)
+        if priority >= self.getpriority(name):
+            del self.attributes[name]
+
+    def __delitem__(self, name):
+        self.delete(name)
```

Comment:
I'm not sure I like the new `delete()` method, seems like users could get confused about their usage and their differences and similarities with `set()`. For keeping the dict-like interface I think we could keep `__delitem__` with a simpler implementation, such as deleting the mapping in `self.attributes`.